### ISSUE TITLE
ETIMEDOUT : API call

### ISSUE THREAD

**Issue by itachi1994** [peter-evans/create-pull-request#733]
### Timeout error for API call

Creates new branch and commits all changes just as expected but fails with the following error:
`_Error: request to https://api.github.com/repos/org/repo/pulls failed, reason: connect ETIMEDOUT_
`
When I run the curl with the token and URL, I get a successful result.

### Steps to reproduce
Runs as a scheduler:
```
 - name: Create Pull Request
      uses: peter-evans/create-pull-request@v3.8.2
      with:
          token: ${{ secrets.ACCESS_PAT }}
          base: master
```

I am unsure if this is a bug or configuration error on my part.

**Comment by peter-evans** [comment]
Hi @itachi1994 

Are you running on self-hosted? A quick search on Google reveals that this kind of error often occurs when behind a corporate proxy.

**Comment by itachi1994** [comment]
Hi, @peter-evans Thank you for the reply. I think this is the case.

**Comment by peter-evans** [comment]
If you found a way to fix it I'm interested to know.

**Comment by itachi1994** [comment]
@peter-evans  I was not able to find a way to fix this issue. I ended up using the "hub" directly.

**Comment by itachi1994** [comment]
@peter-evans  The entire log here:
```
Save persisted git credentials
Determining the base and head repositories
Pull request branch target repository set to repo
Configuring credential for HTTPS authentication
Checking the base repository state
Pull request branch to create or update set to 'test-automate'
Configuring the committer and author
Create or update the pull request branch
Pushing pull request branch to 'origin/test-automate'
  /usr/bin/git push --force-with-lease origin HEAD:refs/heads/test-automate
  remote: 
  remote: Create a pull request for 'test-automate' on GitHub by visiting:        
  remote:      https://github.com/repo-internal/repo/pull/new/test-automate        
  remote: 
  To https://github.com/repo-internal/repo
   * [new branch]      HEAD -> test-automate
Error: request to https://api.github.com/repos/repo-internal/repo/pulls failed, reason: connect ETIMEDOUT 140.82.121.6:443
Restore persisted git credentials
```

**Comment by peter-evans** [comment]
Try setting your corp proxy as an environment variable like this.
```yml
      - name: Create Pull Request
        uses: peter-evans/create-pull-request@v3
        env:
          http_proxy: http://<proxy_address>:<port>
```

**Comment by itachi1994** [comment]
Thank you for the reply. I tried using this, same error.

**Comment by anzboi** [comment]
Very late here, but this error is still here, and is actually a bit of an oversight by the community.

Octokit does not by default look up `https_proxy` environment variables to route its requests. [This actually makes sense](https://github.com/octokit/octokit.js/issues/2098#issuecomment-844673894) in a broader view of what octokit is meant to do, but unfortunately it means most javascript actions simply cannot run on self-hosted runners sitting behind a proxy.

[Octokit has a constructor](https://github.com/octokit/action.js#usage) for use in actions, but it is not widely used, and even if it was, it also does not read proxy env vars yet. Ideally the fix would go in there but it would take a long time to proliferate across javascript actions.

**Comment by peter-evans** [comment]
@anzboi Thank you for investigating! The most widely used constructor for Octokit in actions is probably the one in the actions toolkit here: https://github.com/actions/toolkit/blob/main/packages/github/src/utils.ts
It might help the community to add support for it there.

This actions just uses `octokit/core`, though, because I wanted a little more control over its construction.

I might be able to add the suggestion in this comment to the action.
https://github.com/octokit/octokit.js/issues/2098#issuecomment-844673894

**Comment by peter-evans** [comment]
@anzboi @itachi1994 

I've attempted to add proxy support to the action, but I haven't figured out a good way to test it yet. If either of you are able to test it out, that would be a great help.

You can test it like this using the branch of PR https://github.com/peter-evans/create-pull-request/pull/837
```yml
      - name: Create Pull Request
        uses: peter-evans/create-pull-request@proxy-support
        env:
          https_proxy: https://<proxy_address>:<port>
```

**Comment by peter-evans** [comment]
I figured out a way to test it and added a system test:
https://github.com/peter-evans/create-pull-request-tests/blob/master/.github/workflows/test-command.yml#L964-L988

**Comment by peter-evans** [comment]
Released as `v3.10.0` / `v3`
